code/core/src/csf-tools/CsfFile.ts

```diff
@@ -365,8 +367,12 @@ export class CsfFile {
             ) as t.ImportDeclaration;
             if (importStmt) {
               const { source } = importStmt;
-              if (t.isStringLiteral(source)) {
+              const specifier = importStmt.specifiers.find((spec) => spec.local.name === id);
+              if (t.isStringLiteral(source) && specifier) {
                 this._rawComponentPath = source.value;
+                if (t.isImportSpecifier(specifier) || t.isImportDefaultSpecifier(specifier)) {
+                  this._componentImportSpecifier = specifier;
+                }
```

Comment:
this needs a test change, right?

(I'm also not entirely sure what it does, but that goes for this whole file. 😅 unfortunately the Comment Dealership has raised their prices a lot it seems)

Reply:
I added a comment. This code is tested in the react-docgen.test.ts. It seems like the current CsfFile test don't really test any of the private properties, so it seems more consistent to only test it where this private property is needed.